openpype/hosts/houdini/api/lib.py

```diff
@@ -747,3 +753,69 @@ def get_camera_from_container(container):
 
     assert len(cameras) == 1, "Camera instance must have only one camera"
     return cameras[0]
+
+
+def publisher_show_and_publish(comment=""):
+    """Open publisher window and trigger publishing action."""
+
+    main_window = get_main_window()
+    publisher_window = get_tool_by_name(
+        tool_name="publisher",
+        parent=main_window,
+        reset_on_show=False
+    )
+
+    publisher_window.set_current_tab("publish")
+    publisher_window.make_sure_is_visible()
+    publisher_window.reset_on_show = False
+    publisher_window.set_comment_input_text(comment)
+    publisher_window.reset()
+    publisher_window.click_publish()
+
+
+def self_publish():
+    """Self publish from ROP nodes."""
+
+    current_node = hou.node(".").path()
+
+    host = registered_host()
+    context = CreateContext(host, reset=True)
```

Comment:
Since we already have a CreateContext it would be nice if we could actually pass that on directly to the Publisher UI itself @iLLiCiTiTas we open it. Or is there good reason to not allow that? That could potentially even allow to define custom CreateContext classes (that discover plugins differently?) but use the same UI.